providers/curation/github.js

```diff
@@ -34,86 +37,72 @@ class GitHubCurationService {
-            const sha = masterBranch.data.commit.sha
-            return github.gitdata.createReference({ owner, repo, ref: `refs/heads/${prBranch}`, sha })
-          })
-          .then(() => updatedPatch)
-      })
-      .then(updatedPatch => {
-        const message = `Update ${path} ${coordinates.revision}`
-        if (curationPathSha)
-          return github.repos.updateFile({
-            owner,
-            repo,
-            path,
-            message,
-            content: base64.encode(updatedPatch),
-            branch: prBranch,
-            sha: curationPathSha
-          })
-        return github.repos.createFile({
-          owner,
-          repo,
-          path,
-          message,
-          content: base64.encode(updatedPatch),
-          branch: prBranch
-        })
-      })
-      .then(() => {
-        return github.pullRequests.create({
-          owner,
-          repo,
-          title: this._getPrTitle(coordinates),
-          body: patch.description,
-          head: `refs/heads/${prBranch}`,
-          base: branch
-        })
+  async _writePatch(userGithub, serviceGithub, description, patch, branch) {
+    const { owner, repo } = this.options
+    const coordinates = new EntityCoordinates(patch.coordinates.type, patch.coordinates.provider, patch.coordinates.namespace, patch.coordinates.name)
+    const currentContent = await this.getAll(coordinates)
+    const newContent = patch.revisions
+    const updatedContent = this._updateContent(coordinates, currentContent, newContent)
+    const content = base64.encode(updatedContent)
+    const path = this._getCurationPath(coordinates)
+    const message = `Update ${path}`
+    const committer = userGithub ? await userGithub.users.get({}) : await serviceGithub.users.get({})
+    if (currentContent && currentContent._origin)
+      return serviceGithub.repos.updateFile({
+        owner,
+        repo,
+        path,
+        message,
+        content,
+        branch,
+        sha: currentContent._origin.sha,
+        committer: `{ "name": "${committer.data.name}", "email": "${committer.data.email}" }`,
       })
+    return serviceGithub.repos.createFile({
+      owner,
+      repo,
+      path,
+      message,
+      content,
+      branch,
+      committer: `{ "name": "${committer.data.name}", "email": "${committer.data.email}" }`,
+    })
+  }
+
+  async addOrUpdate(userGithub, serviceGithub, patch) {
+    const { owner, repo, branch } = this.options
+    const master = await serviceGithub.repos.getBranch({ owner, repo, branch: `refs/heads/${branch}` })
+    const sha = master.data.commit.sha
+    const prBranch = await this._getBranchName(userGithub, serviceGithub)
+    await serviceGithub.gitdata.createReference({ owner, repo, ref: `refs/heads/${prBranch}`, sha })
+    await Promise.all(patch.patches.map(throat(1, component => this._writePatch(userGithub, serviceGithub, patch.description, component, prBranch))))
+    // Create the PR using service github object
```

Comment:
we want to create the PR using `userGitHub` so it is attributed to the right person